garnet.py

```diff
@@ -25,25 +25,25 @@
 # set the debug mode to false to speed up construction
 set_debug_mode(False)
 
-
-class Garnet(Generator):
-    def __init__(self, width, height, add_pd, interconnect_only: bool = False,
+def DefineGarnet(width, height, add_pd, interconnect_only: bool = False,
```

Comment:
we use a new pattern for generators: https://github.com/StanfordAHA/gemstone/blob/migrate-magma/gemstone/config_register.py

so before we had

```python
def define_foo(param0, param1):

    class foo(m.Circuit):
        name = f"foo_{param0}_{param1}"
        io = m.IO(...)
        inst = Bar()

    return foo
```

to

```python
class Foo(m.Generator2):
    def __init__(self, param0, param1):
        self.io = m.IO(...)
        self.name = f"Foo_{param0}_{param1}"
        inst = Bar()
```

Reply:
Got it... is this documented anywhere?